octobot/community/supabase_backend/community_supabase_client.py

```diff
@@ -493,14 +489,39 @@ async def _fetch_paginated_history(
                     "timestamp", desc=False
                 )
             )
+
+        return await self.paginated_fetch(
+            client, table_name, request_factory
+        )
+
+    async def paginated_fetch(
+        self,
+        client,
+        table_name: str,
+        request_factory: typing.Callable[
+            [postgrest.AsyncRequestBuilder, postgrest.types.CountMethod], postgrest.AsyncSelectRequestBuilder
+        ],
+    ) -> list:
+        offset = 0
+        max_size = 0
+        total_elements = []
+        request_count = 0
+        total_elements_count = 0
+        while request_count < self.MAX_PAGINATED_REQUESTS_COUNT:
+            request = request_factory(
+                client.table(table_name),
+                None if total_elements_count else postgrest.types.CountMethod.exact
+            )
             if offset:
                 request = request.range(offset, offset+max_size)
-            fetched_elements = (await request.execute()).data
+            result = await request.execute()
+            fetched_elements = result.data
+            total_elements_count = total_elements_count or result.count   # don't change total count within iteration
             total_elements += fetched_elements
             if(
                 len(fetched_elements) == 0 or
                 len(fetched_elements) < max_size or
-                (max_size == 0 and len(fetched_elements) == total_elements_count)
+                (total_elements_count and max_size == 0 and len(fetched_elements) == total_elements_count)
```

Comment:
:+1:

Reply:
I just updated this condition to make it easier to understand

Reply:
:+1: